Document alerts and notifications for NOM 1.6 (#106)
neo4j/docs-ops-manager: modules/ROOT/pages/monitoring/notifications.adoc

= Notifications
:description: This section describes Notifications, Targets and Channels in Neo4j Ops Manager.

> This section describes Notifications and Notification Channels in Neo4j Ops Manager

Reviewer: Caps caps caps, I strongly advise against them.

> Notifications enable forwarding raised alerts over common Channel types to specific Channels. Setting up notifications comprises two steps:

Reviewer: ```suggestion
Notifications enable forwarding raised alerts over common Channel types to specific Channels. 
Setting up notifications comprises two steps:
```

* Channels configuration
* Notifications configuration

The notification management is performed in the *Notifications* section of the NOM settings page.

image::notifications-page.png[width=800]

== Channels

Channel types represent the transport media while Channels represent the actual receivers.

[cols="1,1"]
|===
|Channel type |Channel

|Email (SMTP)
|SMTP Server (NOM user emails)

|Webhook
|Webhook URLs
|===

=== Email (SMTP)

NOM allows configuring a single SMTP Outbox Server with SSL, STARTTLS or no encryption.

> * To configure the outbox server click on *"Configure email"* button from the Channels section to open the creation modal. Fill in the outbox server address and port, the encryption type, the username and password, test the server connection (optional) and click on  *"Confirm"*.

Reviewer: ```suggestion
* To configure the outbox server, use then *"Configure email"* button from the Channels section to open the creation modal. 
Fill in the outbox server address and port, the encryption type, the username and password, test the server connection (optional) and then  *"Confirm"*.
```

[NOTE]
====

> Selecting "None" security for a STARTTLS encrypted server will result in a successful connection test, but will generally fail to send an actual email for the majority of email providers.

Reviewer: ```suggestion
Selecting "None" security for a STARTTLS encrypted server results in a successful connection test, but generally fails to send an actual email for the majority of email providers.
```

====

image::notifications-smtp-configure.png[width=800]

> * To edit the email configuration click again on *"Configure email"*, edit the details and click on  *"Save"*.

Reviewer: ```suggestion
* To edit the email configuration, go back to *"Configure email"*, edit the details and *"Save"*.
```

> * To remove the email configuration click on the *"x"* icon on the grey "Configured Email" tag.

Reviewer: ```suggestion
* To remove the email configuration use the *"x"* icon on the grey "Configured Email" tag.
```

image::notifications-smtp-remove.png[width=800]

=== Webhooks

> NOM allows configuring multiple Webhooks and supports both custom implementations and some common providers (_Discord_, _Teams_ and _Slack_). NOM will forward an alert to a webhook via a HTTP POST request with a JSON payload. A custom webhook implementation should accept the following _Generic_ request body and return a `2XX` status to be rendered successful.

Reviewer: ```suggestion
NOM allows configuring multiple Webhooks and supports both custom implementations and some common providers (_Discord_, _Teams_ and _Slack_). 
NOM forwards an alert to a webhook via a HTTP POST request with a JSON payload. 
A custom webhook implementation should accept the following _Generic_ request body and return a `2XX` status to be rendered successful.
```

._Generic_ webhook payload
[source, json]
----
{
    "title": "Title",
    "description": "Description",
    "severity": "Severity",
    "location": "Location"
}
----

> * To configure a webhook click on *"Add webhook"* button from the Channels section to open the creation modal. Define a name, fill in the webhook type and URL, send a test message (optional) and click on  *"Add"*.

Reviewer: ```suggestion
* To configure a webhook, use the *"Add webhook"* button from the Channels section to open the creation modal. 
Define a name, fill in the webhook type and URL, send a test message (optional), and then  *"Add"*.
```

image::notifications-webhook-add.png[width=800]

* Editing an existing webhook is not supported.
* To remove a webhook click on the *"x"* icon on the grey "Added Webhooks" tag.

image::notifications-webhook-remove.png[width=800]

== Notifications

> Individual Notification Configurations (Notifications) can be configured when at least one channel is setup. A notification defines what type of alerts, in which DBMSs should notify which channel.

Reviewer: ```suggestion
Individual Notification Configurations (Notifications) can be configured when at least one channel is set up.
 A notification defines what type of alerts, in which DBMSs should notify which channel.
```

.Example
[NOTE]
====
When an alert of type [*WARNING*, *ERROR*] is raised inside [*CS Cluster Prod*, *CS Single Prod*] notify NOM users [*nom-admin@neo4j.com*] and webhooks [*Slack#test*, *Generic#prod-errors*].
====

> * To configure a notification click on *"Add notification"* button from the Notifications section to open the creation modal. Define a name, select the DBMSs and alert levels in scope, select the channels and click on  *"Confirm"*.

Reviewer: ```suggestion
* To configure a notification use the *"Add notification"* button from the Notifications section to open the creation modal. 
Define a name, select the DBMSs and alert levels in scope, select the channels, and  *"Confirm"*.
```

[NOTE]
====
Email notifications can only be sent to email addresses of NOM users.
====

image::notifications-notification-configure.png[width=800]

* To edit the notification open the context menu and click on *"Edit"*, update the details and click on  *"Save"*.

image::notifications-notification-edit.png[width=800]

* To remove the notification click on *"Delete"* context menu item.

> * To test the notification click on *"Test"* context menu item. This will send a test message to all configured channels.

Reviewer: ```suggestion
* To test the notification click on *"Test"* context menu item. 
This sends a test message to all configured channels.
```

[IMPORTANT]
====

> Removing all channels that are part of a notification will result in the removal of the notification itself. 

Reviewer: ```suggestion
Removing all channels that are part of a notification results in the removal of the notification itself. 
```
